rosidl_typesupport_cpp/include/rosidl_typesupport_cpp/type_support_map.h

```diff
@@ -15,38 +15,12 @@
 #ifndef ROSIDL_TYPESUPPORT_CPP__TYPE_SUPPORT_MAP_H_
 #define ROSIDL_TYPESUPPORT_CPP__TYPE_SUPPORT_MAP_H_
 
-#include <cstddef>
+// Keep this file not to break API. It was no more than
+// a copy of rosidl_typesupport_c/type_support_map.h
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif
+#warning rosidl_typesupport_cpp/type_support_map.h header is deprecated \
+  in favor of rosidl_typesupport_c/type_support_map.h.
```

Comment:
Hm, I wonder if we should remove this warning for the Foxy backport.  After all, downstream consumers may already be relying on it.  @jacobperron thoughts?

Reply:
I agree, we should avoid backporting deprecation warnings.